src/user-accounts/CreateNewAccount.js

```diff
@@ -31,14 +31,21 @@ export default class CreateNewAccount extends PureComponent {
 
   onRedirectToAccountOpening = () => {
     this.setState({ currency_error: false });
-    if (this.props.account.currency && this.props.account.currency !== '') {
+    if (this.state.selected_currency && this.state.selected_currency !== '') {
```

Comment:
@nazaninreihani this is what you intended right? Cause `this.props.account.currency` will return the currency of the active account.

Reply:
@brucebinary I fixed it here https://github.com/binary-com/binary-next-gen/pull/754/files but this PR seems better to me, 
but the issue I was talking about is not this, the issue is when user chooses a currency in set-currency page or for upgrading, it sends the request to set that currency and in response we should add the currency to account in localStorage, but by moving the refresh code from inside the function which is going to run after setting currency, it's not going to refresh after the response so the currency is added to localStorage but still the app thinks the user has no currency.

Reply:
@nazaninreihani I'll take a look at that. I just tried setting the currency this morning, and I didn't come across any issues.